src/main/java/oncall/controller/AssignController.java

```diff
@@ -0,0 +1,68 @@
+package oncall.controller;
+
+import java.util.function.Supplier;
+import oncall.controller.dto.DateInfo;
+import oncall.controller.dto.MonthAndStartDayOfWeek;
+import oncall.controller.dto.WorkersInfo;
+import oncall.controller.dto.WorkersName;
+import oncall.domain.AssignManager;
+import oncall.domain.Workers;
+import oncall.domain.constants.CustomDayOfWeek;
+import oncall.domain.constants.Month;
+import oncall.view.InputView;
+import oncall.view.OutputView;
+import oncall.view.console.ConsoleWriter;
+
+public class AssignController {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public AssignController(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+        DateInfo dateInfo = getDateInfo();
+        WorkersInfo workersInfo = getWorkersInfo();
+
+        AssignManager assignManager = new AssignManager(
+                workersInfo.weekday(),
+                workersInfo.weekend()
+        );
+        Workers result = assignManager.assign(
+                dateInfo.month(),
+                dateInfo.startDayOfWeek().toDayOfWeek()
+        );
+        outputView.printResult(dateInfo.month(), dateInfo.startDayOfWeek().toDayOfWeek(), result);
+    }
+
+    private DateInfo getDateInfo() {
+        return retry(() -> {
+            MonthAndStartDayOfWeek monthAndStartDayOfWeek = inputView.readMonthAndStartDayOfWeek();
+            Month month = Month.from(monthAndStartDayOfWeek.month());
+            CustomDayOfWeek dayOfWeek = CustomDayOfWeek.from(monthAndStartDayOfWeek.startDayOfWeek());
+            return new DateInfo(month, dayOfWeek);
```

Comment:
해당 부분을 메서드로 뺀 다음에, 메서드 참조를 활용하면 좀 더 가독성에 좋을 것 같습니다, Indent도 하나 덜 수 있고요!

Reply:
잘 이해가 안가는 부분이 있습니다! 
현재 코드에서 호출하는 함수들이 모두 각기 다른 객체를 반환하고 있는데, 메서드로 뺀다는 말씀은 세 줄의 코드를 아래와 같이 수정하라는 말씀일까요? 

```suggestion
            MonthAndStartDayOfWeek monthAndStartDayOfWeek = inputView.readMonthAndStartDayOfWeek();
            
            return getDateInfo(monthAndStartDayOfWeek);
            
     public DateInfo getDateInfo(MonthAndStartDayOfWeek monthAndStartDayOfWeek) {
            Month month = Month.from(monthAndStartDayOfWeek.month());
            CustomDayOfWeek dayOfWeek = CustomDayOfWeek.from(monthAndStartDayOfWeek.startDayOfWeek()); 
            
            return new DateInfo(month, dayOfWeek);
     } 
```

Reply:
`retry`의 람다 함수 내부에 하는 일이 많다는 뜻이었어요. 
저 네 줄을 하나의 메서드로 빼 둔 다음에,
```java
private DateInfo constructDateInfo() {
    MonthAndStartDayOfWeek monthAndStartDayOfWeek = inputView.readMonthAndStartDayOfWeek();
    Month month = Month.from(monthAndStartDayOfWeek.month());
    CustomDayOfWeek dayOfWeek = CustomDayOfWeek.from(monthAndStartDayOfWeek.startDayOfWeek());
    return new DateInfo(month, dayOfWeek);
}
```

`retry`에서 메서드 참조를 통해 가독성을 좀 더 높일 수 있어보입니다
```java
private DateInfo getDateInfo() {
    return retry(this::constructDateInfo);
}
```

Reply:
친절한 설명 감사드립니다! 반영하여 수정하겠습니다 😊😊